prow/plugins/lgtm/lgtm.go

```diff
@@ -300,12 +330,43 @@ func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent, log *logrus
 		return nil
 	}
 
-	// Don't bother checking if it has the label...it's a race, and we'll have
-	// to handle failure due to not being labeled anyway.
 	org := pe.PullRequest.Base.Repo.Owner.Login
 	repo := pe.PullRequest.Base.Repo.Name
 	number := pe.PullRequest.Number
 
+	// Check if we have LGTM label
+	hasLGTM := false
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repo, number)
+	}
+	hasLGTM = github.HasLabel(lgtmLabel, labels)
+
+	if hasLGTM == true {
+		// Get combined status to see if we already know a lgtm tree-hash
+		combined, err := gc.GetCombinedStatus(org, repo, *pe.PullRequest.MergeSHA)
```

Comment:
This won't be the right SHA because the PR just changed. This is the second point I mentioned here: https://github.com/kubernetes/test-infra/issues/6353#issuecomment-412197985

Reply:
Please read further, as I store the tree SHA inside the description of the status... That one doesn't change according to @thockin.

Reply:
Sorry, it wasn't clear on the mobile phone... please discard my previous comment. I will check if I can use another ref (The ref can be a SHA, a branch name, or a tag name.).

Reply:
The problem isn't with the ref, it is with the use of status contexts. Using a status context associates the state you want to store with a commit SHA that will necessarily change after a rebase. At that time you won't know what the previous SHA was so you won't be able to look up the state you stored.